synfig-core/src/synfig/valuenodes/valuenode_log.cpp

```diff
@@ -120,7 +122,22 @@ ValueNode_Logarithm::operator()(Time t)const
 	DEBUG_LOG("SYNFIG_DEBUG_VALUENODE_OPERATORS",
 		"%s:%d operator()\n", __FILE__, __LINE__);
 
-	Real link     = (*link_)    (t).get(Real());
+	ValueBase input_value = (*link_)(t);
+    Real link = 0;
+
+	switch (input_value.get_type())
+	{
+		case ValueBase::TYPE_REAL:
```

Comment:
It won't work. `ValueBase::get_type()` doesn't return an enumeration.
Please check other valuenode implementations to see how it should be done.